### PR TITLE
Add shellcheck

### PR DESCRIPTION
**Pull request by sebrandon1**
Tests incoming PRs against shellcheck.  Watches the shell files in the `hack` folder.

Similar to: https://github.com/test-network-function/cnf-certification-test/pull/185

### CODE REVIEW

File: .github/workflows/pre-main.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-22.04
```
Comment (josephdrichard) [inline .github/workflows/pre-main.yaml:13]: Why Ubuntu and not rhel or fedora?

**Review by josephdrichard — COMMENTED** [review]
Is there a bug this is linked to?

File: .github/workflows/pre-main.yaml (hunk 1 above)
Comment (sebrandon1) [reply .github/workflows/pre-main.yaml:13]: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

**Pull request closed without merging**